break.js

```diff
@@ -1,32 +1,44 @@
 var utils = require('./utils')
+var ext = require('./bn-extensions')
 
 // break utxos into the maximum number of 'output' possible
 module.exports = function broken (utxos, output, feeRate) {
-  if (!isFinite(utils.uintOrNaN(feeRate))) return {}
+  if (!isFinite(utils.bnOrNaN(feeRate))) return {}
 
   var bytesAccum = utils.transactionBytes(utxos, [])
-  var value = utils.uintOrNaN(output.value)
+  var value = utils.bnOrNaN(output.value)
   var inAccum = utils.sumOrNaN(utxos)
+
   if (!isFinite(value) ||
-      !isFinite(inAccum)) return { fee: feeRate * bytesAccum }
+      !isFinite(inAccum)) return { fee: feeRate.mul(bytesAccum) }
 
   var outputBytes = utils.outputBytes(output)
-  var outAccum = 0
+  var outAccum = ext.BN_ZERO
   var outputs = []
 
   while (true) {
-    var fee = feeRate * (bytesAccum + outputBytes)
+    // feeRate * (bytesAccum + outputBytes)
+    var fee = ext.multiply(feeRate, ext.add(bytesAccum, outputBytes))
+
+    var outputTotal = ext.add(fee, value)
+    var totalSoFar = ext.add(outAccum, outputTotal)
+    var isLessThanInAccumulator = ext.lessThan(inAccum, totalSoFar)
 
     // did we bust?
-    if (inAccum < (outAccum + fee + value)) {
+    // inAccum < (outAccum + fee + value)
+    if (isLessThanInAccumulator) {
```

Comment:
can we just do `ext.lessThan(inAccum, ext.add(outAccum, fee, value))`

you are gonna have to make `add` support more than 2 arguments

Reply:
also probably want to do `lt` instead of `lessThan`